controllers/nstemplateset/nstemplateset_controller.go

```diff
@@ -176,13 +175,17 @@ func (r *Reconciler) deleteNSTemplateSet(logger logr.Logger, nsTmplSet *toolchai
 	username := nsTmplSet.GetName()
 
 	// delete all namespace one by one
-	deletedAny, err := r.namespaces.delete(logger, nsTmplSet)
-	if err != nil || deletedAny {
+	allDeleted, err := r.namespaces.ensureDeleted(logger, nsTmplSet)
+	// when err, status Update will not trigger reconcile, sending returning error.
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if !allDeleted {
 		return reconcile.Result{}, nil
```

Comment:
Let's add some comments here.
```suggestion
		// One or more namespaces is not yet deleted. We can stop here. When it's finally deleted it will trigger another reconcile.
		return reconcile.Result{}, nil
```

Reply:
done